scipy/_lib/_array_api.py

```diff
@@ -163,3 +163,45 @@ def copy(x, *, xp=None):
         xp = array_namespace(x)
 
     return as_xparray(x, copy=True, xp=xp)
+
+
+def assert_equal(actual, desired, err_msg='', xp=None):
+    # TODO compare arrays from different libraries
+    if xp is None:
+        xp = array_namespace(actual)
+    if 'cupy' in xp.__name__:
+        return xp.testing.assert_array_equal(actual, desired, err_msg=err_msg)
+    elif 'torch' in xp.__name__:
+        return xp.testing.assert_close(actual, desired, rtol=0, atol=0,
+                                       msg=err_msg)
+    return np.testing.assert_array_equal(actual, desired, err_msg=err_msg)
+
+
+def assert_close(actual, desired, rtol=1e-07, atol=0, err_msg='', xp=None):
+    # TODO compare arrays from different libraries
```

Comment:
can you remove this comment? I think we're going the way of comparing arrays from the same library here, right?

Reply:
doh, you wrote that an hour ago.